compiler/src/iree/compiler/Codegen/TransformDialectStrategies/GPU/Common.cpp

```diff
@@ -549,6 +551,15 @@ Value mlir::iree_compiler::gpu::buildConvertToTensorCoreOp(
       config.unrollVectorsGpuWmma = true;
     b.create<ApplyPatternsOp>(funcH, config);
   }
+
+  Value forH = b.create<transform::MatchOp>(
+      transform::OperationType::get(b.getContext(), "scf.for"), funcH,
+      b.getStrArrayAttr({scf::ForOp::getOperationName()}),
+      /*matchInterfaceEnum=*/transform::MatchInterfaceEnumAttr(),
+      /*opAttrs=*/DictionaryAttr(),
+      /*filterResultType=*/TypeAttr());
+  b.create<SynchronizeLoopOp>(forH);
```

Comment:
This transform is required for correctness, right?
How do we make sure we don't forget about it if we were to do a similar transform somewhere else?

Put differently, shouldn't the barrier be introduced by the optimization that needs it in the first place?

Then we can just keep the later `EliminateGpuBarriersOp`.

Reply:
Great point, I'll create a specific IREE transform to perform this and wrap in barriers.
This requires an upstream refactoring that will take time to refactor so just adding a TODO in this PR for now.
Once integrate has gone through I can finalize.